src/Microsoft.ML.Core/Data/ITransformModel.cs

```diff
@@ -54,5 +54,10 @@ public interface ITransformModel
         /// <returns>The transform model as an <see cref="IRowToRowMapper"/>. If not all transforms
         /// in the pipeline are <see cref="IRowToRowMapper"/> then it returns null.</returns>
         IRowToRowMapper AsRowToRowMapper(IExceptionContext ectx);
+
+        /// <summary>
+        /// Get the loader information from the model.
+        /// </summary>
+        IDataView GetLoader();
```

Comment:
Hi @zeahmed , thanks for beginning to look at this.

I don't consider this an appropriate addition to `ITransformModel`. Transform models are for, well, transforms. :) There's also absolutely no reason to believe that a loader will be involved *at all*, since many usages of source data are programmatic.

The problem with the current API is that it did not account for the presence of a loader-model *at all*, and this change does not seem to address that core issue.